### PR TITLE
moved all the relevant solver-related test cases to test_solvers.py

### PR DESCRIPTION
**Pull request by sksingh1202**
**References to other Issues or PRs:**
Partial fix for #18666 

#### Brief description of what is fixed or changed
moved all the solver - related test cases from the existing files test_matrices.py and test_commonmatrix.py to a new file test_solvers.py.

#### Release Notes

- matrices
  -  Added test_solvers.py file which contains all the solver-related test cases.

### CODE REVIEW

File: sympy/matrices/tests/test_matrices.py [hunk 1]
```diff
@@ -597,33 +597,6 @@ def test_issue_17247_expression_blowup_30():
...
-#         [    0, 1 - x, x + 1,     0],
-#         [    0,     0,     0, x + 1]])
-#     assert M.LDLsolve(ones(4, 1)) == Matrix([
-#         [(x + 1)/(4*x)],
-#         [(x - 1)/(4*x)],
-#         [(x + 1)/(4*x)],
-#         [    1/(x + 1)]])
-
-@XFAIL # dotprodsimp is not on by default in this function
-def test_issue_17247_expression_blowup_32():
```
Comment (mohitacecode) [inline sympy/matrices/tests/test_matrices.py]: You can ignore the tests like `expression_blowup`.

**Comment by mohitacecode** [comment]
there are conflicts have you updated (`git fetch upstream`) the branch first before doing changes.

**Comment by asmeurer** [comment]
I the past there have been issues with having two files with the same name in the repo. I don't know if those issues still exist, but it's worth checking if there are any problems, since we already have sympy/solvers/tests/test_solvers.py .

File: sympy/matrices/tests/test_matrices.py (hunk 1 above)
Comment (sksingh1202) [reply sympy/matrices/tests/test_matrices.py]: Thank you very much for the suggestion.

**Comment by sksingh1202** [comment]
Thank you, @mohitacecode for mentioning, I hadn't pulled the changes before. Now I have pulled the changes from my fork to my local repository. Do I have to pull the changes from the sympy/sympy repository too? Can I update the pull request somehow or do I have to open a new pull request after fetching the changes? I am new to open source and it would be great if someone could help me out.
Thank you.

**Comment by sksingh1202** [comment]
@asmeurer , thank you for pointing this out. I would check this and then possibly rename the file.

**Comment by sylee957** [comment]
There are already `test_matrices` in assumptions test, but it does not have any problem for now.
I think that it should be same for here.

**Comment by mohitacecode** [comment]
> Thank you, @mohitacecode for mentioning, I hadn't pulled the changes before. Now I have pulled the changes from my fork to my local repository. Do I have to pull the changes from the sympy/sympy repository too? Can I update the pull request somehow or do I have to open a new pull request after fetching the changes? I am new to open source and it would be great if someone could help me out.
> Thank you.

Try to pull the changes from `sympy` repository and when you will merge those changes you will get a merge conflict in`sympy/matrices/tests/test_matrices.py` solve it and then proceed.

**Comment by sksingh1202** [comment]
Thank you for your help and sorry for the delay as I was unable to give much time to this because of having exams now.

**Comment by sylee957** [comment]
Unfortunately, #18793 had arrived to the solution faster, so I may close this.

**Pull request closed without merging**